src/renderer/components/+workloads-statefulsets/statefulset-scale-dialog.tsx

```diff
@@ -16,24 +16,31 @@ interface Props extends Partial<DialogProps> {
 
 @observer
 export class StatefulSetScaleDialog extends Component<Props> {
-  @observable static isOpen = false;
-  @observable static data: StatefulSet = null;
+  static metadata = observable({
+    isOpen: false,
+    data: null as StatefulSet,
+  });
 
   @observable ready = false;
   @observable currentReplicas = 0;
   @observable desiredReplicas = 0;
 
+  constructor(props: Props) {
+    super(props);
+    makeObservable(this);
+  }
+
   static open(statefulSet: StatefulSet) {
-    StatefulSetScaleDialog.isOpen = true;
-    StatefulSetScaleDialog.data = statefulSet;
+    StatefulSetScaleDialog.metadata.isOpen = true;
+    StatefulSetScaleDialog.metadata.data = statefulSet;
```

Comment:
FYI you could have kept these like this with static getters too

Reply:
And keep the state in some `static data = observable.box({})`?  It would be more cumbersome imho..
Or did you mean something different?

Reply:
I mean the following:

```ts
static metadata = observable({
	  isOpen: false,
	  data: null as StatefulSet,
});

get isOpen() {
	return StatefulSetScaleDialog.metadata.isOpen;
}

set isOpen(val) {
	StatefulSetScaleDialog.metadata.isOpen = val;
}

get data() {
	return StatefulSetScaleDialog.metadata.data;
}

set data(val) {
	StatefulSetScaleDialog.metadata.data = val;
}
```

This way you change this top thing and not have all the other changes sprinkled throughout the file. Just a thought